include/cutlass/gemm/collective/xe_mma.hpp

```diff
@@ -146,6 +147,75 @@ struct CollectiveMma<MainloopIntelXeXMX16<Stages, Schedule>, TileShape_, Element
     return Params{tiled_copy_a, tiled_copy_b};
   }
 
+  template <class EngineIn,
+      class EngineOut,
+      class LayoutIn,
+      class LayoutOut,
+      class... Ts>
+  CUTLASS_DEVICE
+  void convert_FP8_to_FP16(
+          Tensor<EngineIn, LayoutIn> const& in,
+          Tensor<EngineOut, LayoutOut>& out) {
+
+    static_assert(is_rmem<EngineIn>::value, "Input tensor for A conversion must come from registers");
+    static_assert(is_rmem<EngineOut>::value, "Output tensor for A conversion must come from registers");
+    static_assert(cosize_v<LayoutIn> == cosize_v<LayoutOut>);
+    static_assert(size_v<LayoutIn> == cosize_v<LayoutIn>);
+    static_assert(size_v<LayoutOut> == cosize_v<LayoutOut>);
+
+    using SrcType = typename EngineIn::value_type;
+    using DstType = typename EngineOut::value_type;
+
+    static_assert(std::is_same_v<SrcType, uint8_t>, "Expected fp8 (E4M3) input as uint8_t");
+    static_assert(std::is_same_v<DstType, half_t>, "Expected fp16 output as half_t");
+
+    auto const& src = in(_, _, _);
+    auto const& dst = out(_, _, _);
+
+    SrcType const* pSrc = src.data();
+    DstType* pDst = dst.data();
+
+    constexpr int num_elements = decltype(size(src))::value;
+    constexpr int vec_size = 16;
+    constexpr int iters = num_elements / vec_size;
+    using SrcArray = cutlass::Array<uint8_t, vec_size>;
+    using DstArray = cutlass::Array<uint16_t, vec_size>;
+    // TODO(Codeplay): Move conversion to NumericArrayConverter
+    // We don't know why the compiler only vectorizes the conversion
+    // when we use an array of size 16, instead of simply converting
+    // each element in a loop because these arrays are separate for each work-item.
+    // To clarify, since each array is created for a work-item's elements,
+    // the computation corresponding to the elements belonging to one work item
+    // is not implicitly vectorized across a subgroup.
+    // Instead, one element of each work-item would on a separate lane,
+    // and all lanes would be in convergence.
+    // Somehow the compiler is unable to determine convergence without
+    // such a workaround.
+    // Ref: https://github.com/codeplaysoftware/cutlass-sycl/pull/284#discussion_r2020421508
```

Comment:
Hi @rolandschulz, can you please confirm if this note is correct? Thanks

`E4M3_to_FP16_vec16` or `E5M2_to_FP16_vec16` do not seem to be _explicitly_ vectorized. 
The 16-element-arrays are created per work-item (i.e. they are in registers of individual work items), so for one work item, all the 16 elements in the array should map to the same vector lane in the hardware.

Is it the case that the compiler somehow identifies convergence & implicitly vectorizes `E4M3_to_FP16_vec16` in such a manner that each lane simultaneously performs computation for one element of each array of each work item?

Thanks!

Reply:
No. If we want SIMD16 we should just do one scalar operation per work-item. All our kernels run with subgroup size of 16. The compiler should turn the 16 scalar operations into SIMD16. If that doesn't happen email me.

Reply:
Thanks for your response, @rolandschulz!

> No

Sorry for my incoherent comments! The rest of your response seem to be in line with what I tried to convey (i.e. tiling the loop by creating arrays of size 16 should not have been required since each element of a work-item array map to just one SIMD lane). As you advised, I'll reach out to you.

Thanks again!